### PR TITLE
Lua SLAM replacement

### PR DESCRIPTION
**Pull request by Ultimeit**
No one asked for that, no one even bothers using SLAM's cuz it's one of the most useless weapons in Horde currently, but I did it ʕ•ᴥ•ʔ
On a bit more serious note, this replaces built-in npc_tripmine upon placement with a bit more PvE oriented counterpart.
Only owner(or NPC's) can detonate it by dealing damage, also it's getting triggered only by not player-owned NPC's. Damage/Explosion radius is configurable to be further balanced. Laser length is limited, because no one needs lasers that cross whole map. It also can be defused(ConVar horde_tripmine_reusable) by owner(or by anyone if owner disconnected), if it has correct class, by pressing +use on it.
P.S.I wonder how much zombies I could've killed today instead of spending so much time on researching useless SLAM mechanics...

### CODE REVIEW

**Comment by Ultimeit** [comment]
That reminds me, that current engineer turrets are NPC's too and guess they can detonate mines by dealing damage to it >.>

**Review by tpan496 — CHANGES_REQUESTED** [review]
This is some good stuff; I almost forgot SLAM exists. Have some small style comments.

File: entities/entities/npc_tripmine_horde/init.lua [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+			--ent2:SetOwner(owner)
+			--ent2:SetHordeOwner(owner)
+			ent2.HordeOwner = owner
+			ent2:Spawn()
+		end)
+	end
+end)
+
+local reuse = CreateConVar("horde_tripmine_reusable", 1, FCVAR_SERVER_CAN_EXECUTE, "Can tripwire mines(SLAM's) be picked up after being armed?")
+local defuse = Sound("weapons/c4/c4_disarm.wav")
```
[the remark this answers is not in the document]
Comment (Ultimeit) [reply entities/entities/npc_tripmine_horde/init.lua:64]: Afaik, missing sounds doesn't cause errors. Though this probably should be changed to some kind of HL2 sound.

File: entities/entities/npc_tripmine_horde/cl_init.lua [hunk 2]
```diff
@@ -0,0 +1,18 @@
+include("shared.lua")
+killicon.AddAlias("npc_tripmine_horde", "npc_tripmine")
+
+local lmat = Material("trails/laser")
+local color_armed, color_enemy = Color(0, 200, 200), Color(255, 0, 0)
+function ENT:Draw()
+	self:DrawModel()
+end
+
```
[the remark this answers is not in the document]
Comment (Ultimeit) [reply entities/entities/npc_tripmine_horde/cl_init.lua:9]: I wouldn't mind if you'll replace tabs into spaces, however stopping using tabs for me would be problematic. That's pretty old habit for me already.

File: entities/entities/npc_tripmine_horde/init.lua [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+function ENT:Use(ply)
+	local owner = self:GetHordeOwner()
+	if reuse:GetBool() == false then return end
+	if self:GetTriggered() then return end
+	if IsValid(owner) and owner != ply then return end	-- feel free to steal disconnected player mines
+
+	-- I don't want player to be able to difuse mines with wrong class >:c
+	-- btw, saboteur class when?
```
[the remark this answers is not in the document]
Comment (Ultimeit) [reply entities/entities/npc_tripmine_horde/init.lua:72]: Ugh, sadly no one ported Izaro to gmod yet, guess we should delay saboteur ascendancy then.

**Comment by Ultimeit** [comment]
Also, 2.4x damage of RPG(10$ per shot) hit with estimate same AoE radius doesn't sounds that impressive for a mine(100$ per mine). Guess either RPG ammo or SLAM ammo price should be rebalanced. 
Also, wouldn't mind if I'll make Incendiary grenade to use different ammo than HL2 grenades? Guess that's pretty obvious that they are way too powerful even for 100$. Also their ammo can be duped infinitely by playing demolition class and constantly dropping HL2 grenade and receiving a new one from perk, gonna post that to issues list as well as possible fix.

File: entities/entities/npc_tripmine_horde/cl_init.lua (hunk 2 above)
[the remark this answers is not in the document]
Comment (tpan496) [reply entities/entities/npc_tripmine_horde/cl_init.lua:6]: For this PR, this is fine. However, if you are making changes to existing files you **have** to use spaces. Not sure which editor you are using, but on VS code it doesn't any effort to do that (just a simple toggle on the top right corner), and even if you are pressing tabs the editor still generates spaces.

**Comment by tpan496** [comment]
LGTM 👍

**Pull request merged**